clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2653,6 +2653,25 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
   }
   O << "};\n\n";
 
+  O << "// array representing kernel parameters used in all kernels defined in "
+       "the\n";
+  O << "// corresponding source\n";
+  O << "static constexpr\n";
+  O << "const bool kernel_param_used[] = {\n";
```

Comment:
This [example](https://jira.devtools.intel.com/browse/CMPLRLLVM-21326?focusedCommentId=8464268&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-8464268) had kernel params marked as USED, so I chose `kernel_param_used` instead of `param_omittable`